Sources/LeafKit/DocumentLoader.swift

```diff
@@ -0,0 +1,204 @@
+struct UnresolvedDocument {
+    let name: String
+    let raw: [Syntax]
+    
+    var unresolvedDependencies: [String] {
+        return extensions.map { $0.key }
+    }
+    
+    private var extensions: [Syntax.Extend] {
+        return raw.compactMap {
+            switch $0 {
+            case .extend(let e): return e
+            default: return nil
+            }
+        }
+    }
+}
+
+struct ResolvedDocument {
+    let name: String
+    let ast: [Syntax]
+    
+    init(name: String, ast: [Syntax]) throws {
+        for syntax in ast {
+            switch syntax {
+            case .extend, .export:
+                // all extentions MUST be resolved
+                throw "unresolved ast"
+            default:
+                continue
+            }
+        }
+        
+        self.name = name
+        self.ast = ast
+    }
+}
+
+final class DocumentLoader {
+    private var fileAccess: FileAccessProtocol
```

Comment:
I assumed that it would have some sort of nio thing in the end.. I have it setup this way because I thought maybe the protocol could allow the user to do more flexibility if they want and default fileio.

I am not sure which is right here, I've gone back and forth a lot

Reply:
keeping this as a test only for now, moving to extend resolver and punting this work to the renderer 👍